src/core/utils/misc.cpp

```diff
@@ -241,6 +242,11 @@ int Utils::Misc::pythonVersion()
         python_proc.start("python3", QStringList() << "--version", QIODevice::ReadOnly);
         if (python_proc.waitForFinished()) {
             if (python_proc.exitCode() == 0) {
+                QByteArray output = python_proc.readAllStandardOutput();
+                if (output.isEmpty())
+                    output = python_proc.readAllStandardError();
+                const QByteArray version_str = output.split(' ').last();
+                Logger::instance()->addMessage(QCoreApplication::translate("misc", "Python version: %1").arg(QString(version_str)), Log::INFO);
                 version = 3;
```

Comment:
You are assuming if executable is called `python3` then `version = 3`. Most times will be correct but now we can check the exact version. Why you don't call `pythonVersionComplete()` to be sure?

In Linux you are looking for `python3` and `python3` but not for `python`. I know that pythonX should exists.. but there are a ton of distributions... Furthermore, the code to check python3 and python2 is the same so you can have a loop:

versions = [python3, python2, python]
for ( version in versions ) {}
->check if the executable exists
->extract the version with `pythonVersionComplete()`